snuba/subscriptions/executor_consumer.py

```diff
@@ -200,11 +204,24 @@ def __init__(
         # Commit is only passed here because we are temporarily
         # skipping executions during the transition phase.
         commit: Callable[[Mapping[Partition, Position]], None],
+        partitions: Mapping[Partition, int],
     ) -> None:
         self.__dataset = dataset
         self.__entity_names = set(entity_names)
+
+        if state.get_config("use_calculated_max_concurrent_queries", False):
+            calculated_max_concurrent_queries = (
```

Comment:
A comment explaining this equation (why 64?) would be good.

Reply:
If 64 is meant to represent the total number of partition for the topic, that will be a problem. 
Not all topics have the same number of partitions. The errors executor topic has 32 for example, but smaller ones may have less and the errors one is likely going to be increased. 
So that cannot be hardcoded.

The easy way is to add it in settings per executor. It is not great because we would have another manual operation to fix the config after adding partitions, which makes things less autonomous.

The right way is to use the Kafka admin api to get the current number of partitions for the consumer when you subscribe to a topic.

Reply:
addressed in https://github.com/getsentry/snuba/pull/2888